src/configure/mod.rs

```diff
@@ -606,29 +629,38 @@ fn try_imap_one_param(context: &Context, param: &LoginParam) -> Option<bool> {
 
 fn try_smtp_connections(
     context: &Context,
-    mut param: &mut LoginParam,
+    param: &mut LoginParam,
+    original_param: &LoginParam,
     was_autoconfig: bool,
 ) -> bool {
     /* try to connect to SMTP - if we did not got an autoconfig, the first try was SSL-465 and we do a second try with STARTTLS-587 */
-    if let Some(res) = try_smtp_one_param(context, &param) {
+    if let Some(res) = try_smtp_one_param(context, param) {
         return res;
     }
     if was_autoconfig {
         return false;
     }
     progress!(context, 850);
-    param.server_flags &= !(DC_LP_SMTP_SOCKET_FLAGS as i32);
-    param.server_flags |= DC_LP_SMTP_SOCKET_STARTTLS as i32;
-    param.send_port = 587;
+    if original_param.server_flags & (DC_LP_SMTP_SOCKET_FLAGS as i32) == 0 {
+        param.server_flags &= !(DC_LP_SMTP_SOCKET_FLAGS as i32);
+        param.server_flags |= DC_LP_SMTP_SOCKET_STARTTLS as i32;
+    }
+    if original_param.send_port == 0 {
+        param.send_port = 587;
+    }
 
-    if let Some(res) = try_smtp_one_param(context, &param) {
+    if let Some(res) = try_smtp_one_param(context, param) {
         return res;
     }
     progress!(context, 860);
-    param.server_flags &= !(DC_LP_SMTP_SOCKET_FLAGS as i32);
-    param.server_flags |= DC_LP_SMTP_SOCKET_STARTTLS as i32;
-    param.send_port = 25;
-    if let Some(res) = try_smtp_one_param(context, &param) {
+    if original_param.server_flags & (DC_LP_SMTP_SOCKET_FLAGS as i32) == 0 {
```

Comment:
I first wanted to do some flag refactoring that does not need so much bit operations but then it seemed too risky to me because a bug here may easily stay unnoticed but will keep away new users.

Also, does someone know why DC_LP_SMTP_SOCKET_FLAGS is usize? It is casted to i32 everywhere it is used.